File: common/src/main/java/example/common/domain/AgeConverter.java
package example.common.domain;

import org.seasar.doma.ExternalDomain;
import org.seasar.doma.jdbc.domain.DomainConverter;

@ExternalDomain
public class AgeConverter implements DomainConverter<Age, Integer> {
  @Override
  public Integer fromDomainToValue(Age age) {
    return age.value();
  }

  @Override
  public Age fromValueToDomain(Integer value) {
    return value == null ? null : new Age(value);
  }
}

File: dsl-style-kotlin/src/main/kotlin/example/dsl/style/kotlin/domain/AgeConverter.kt
package example.dsl.style.kotlin.domain

import org.seasar.doma.ExternalDomain
import org.seasar.doma.jdbc.domain.DomainConverter

@ExternalDomain
class AgeConverter : DomainConverter<Age?, Int?> {
    override fun fromDomainToValue(age: Age?): Int? {
        return age?.value
    }

    override fun fromValueToDomain(value: Int?): Age? {
        return value?.let { Age(it) }
    }
}
